engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/hashing/typed/addonlyopen/gen/TypedHashDispatcher.java

```diff
@@ -0,0 +1,64 @@
+import java.util.Arrays;
+
+/**
+ * The TypedHashDispatcher returns a pre-generated and precompiled hasher instance suitable for the provided column
+ * sources, or null if there is not a precompiled hasher suitable for the specified sources.
+ */
+public class TypedHashDispatcher {
+    private TypedHashDispatcher() {
+        // static use only
+    }
+
+    public static AddOnlyUpdateByStateManagerTypedBase dispatch(ColumnSource[] tableKeySources,
+            ColumnSource[] originalTableKeySources, int tableSize, double maximumLoadFactor,
+            double targetLoadFactor) {
+        final ChunkType[] chunkTypes =
+                Arrays.stream(tableKeySources).map(ColumnSource::getChunkType).toArray(ChunkType[]::new);;
+        if (chunkTypes.length == 1) {
+            return dispatchSingle(chunkTypes[0], tableKeySources, originalTableKeySources, tableSize, maximumLoadFactor,
+                    targetLoadFactor);
+        }
+        return null;
+    }
+
+    private static AddOnlyUpdateByStateManagerTypedBase dispatchSingle(ChunkType chunkType,
+            ColumnSource[] tableKeySources, ColumnSource[] originalTableKeySources, int tableSize,
+            double maximumLoadFactor, double targetLoadFactor) {
+        switch (chunkType) {
+            default:
+                throw new UnsupportedOperationException("Invalid chunk type for typed hashers: " + chunkType);
+            case Char:
+                return new AddOnlyUpdateByHasherChar(tableKeySources, originalTableKeySources, tableSize,
+                        maximumLoadFactor, targetLoadFactor);
+            case Byte:
+                return new AddOnlyUpdateByHasherByte(tableKeySources, originalTableKeySources, tableSize,
+                        maximumLoadFactor, targetLoadFactor);
+            case Short:
+                return new AddOnlyUpdateByHasherShort(tableKeySources, originalTableKeySources, tableSize,
+                        maximumLoadFactor, targetLoadFactor);
+            case Int:
+                return new AddOnlyUpdateByHasherInt(tableKeySources, originalTableKeySources, tableSize,
+                        maximumLoadFactor, targetLoadFactor);
+            case Long:
+                return new AddOnlyUpdateByHasherLong(tableKeySources, originalTableKeySources, tableSize,
+                        maximumLoadFactor, targetLoadFactor);
+            case Float:
+                return new AddOnlyUpdateByHasherFloat(tableKeySources, originalTableKeySources, tableSize,
+                        maximumLoadFactor, targetLoadFactor);
+            case Double:
+                return new AddOnlyUpdateByHasherDouble(tableKeySources, originalTableKeySources, tableSize,
+                        maximumLoadFactor, targetLoadFactor);
+            case Object:
+                return new AddOnlyUpdateByHasherObject(tableKeySources, originalTableKeySources, tableSize,
+                        maximumLoadFactor, targetLoadFactor);
```

Comment:
For the final version, we might want to do the double type dispatching so that those versions don't need to wait for compilation.